src/Model/Item.php

```diff
@@ -3483,11 +3483,8 @@ private static function addVisualAttachments(PostMedias $PostMedias, array $item
 				continue;
 			}
 
-			if ($PostMedia->mimetype->type == 'image') {
-				$preview_size = $PostMedia->width > $PostMedia->height ? Proxy::SIZE_MEDIUM : Proxy::SIZE_LARGE;
```

Comment:
This line was meant to avoid problems with pictures that are extremely long but really small. I think that we still should handle this.

Reply:
Yes, but not at this point. Otherwise it's forcing preview images that we generate ourselves inside a 640px square to be upscaled, which is even less ideal.

Reply:
Yes. But we could check for some really dramatic width to height ratios.

Reply:
Even then, if the preview has been generated inside that 640px box, then changing the display size will upscale them as dramatically as they aspect ratio. We need to change the preview scaling code to limit just the width to 640px in order for both landscape and portrait preview images to look fine inside posts.